### DISCUSSION TITLE
Should buildStart run?

### DISCUSSION THREAD

**Discussion by wmertens** [vitest-dev/vitest#5479] [Q&A] [answered]
When using rollup-plugin-node-externals with vitest, it crashes on [this line](https://github.com/Septh/rollup-plugin-node-externals/blob/main/source/index.ts#L131).
The reason is that the `include` array wasn't initialized, which happens inside `buildStart`.

Is buildStart not supposed to run always?

**Comment by hi-ogawa** [comment]
I think this tends to happen when using Vite's `createServer` API but forgetting to run `server.pluginContainer.buildStart`. For example, vite-node has this readme:

https://github.com/vitest-dev/vitest/blob/e4e939ba2ab48c67ee14b82ec957fc9a8a52756c/packages/vite-node/README.md?plain=1#L87-L95

I don't think Vitest has any issues with this though. Can you provide a reproduction?

**Reply by hi-ogawa** [reply, marked as answer]
Just did a quick check and it looks working to me:

https://stackblitz.com/edit/vitest-dev-vitest-xsr7nz?file=vite.config.ts

<details><summary>Show screenshot</summary>

![image](https://github.com/vitest-dev/vitest/assets/4232207/1a7c51ff-bf76-4a21-b936-1a1392210f67)

</details>

**Reply by wmertens** [reply]
apologies for not responding any more. I don't remember what the problem was, but indeed it wasn't buildStart and everything's ok now. Thanks!

**Reply by hi-ogawa** [reply]
Actually what you were seeing was probably right. Vitest had some issue with `rollup-plugin-node-externals` but it has been fixed sometime ago
- https://github.com/vitest-dev/vitest/issues/5616
